### ISSUE TITLE
Edit island neighbors values in spatial Wobject Pysal

### ISSUE THREAD

**Issue by MYMahfouz** [pysal/libpysal#303]
I am working on clustering a map with some islands in it.
These islands have no neighbors in the w object and so I can't run the maxp clustering function. I get the command ''No initial solution found''.
What I tried to do is to calculate the neighbors using the distance spatial weights, then substituting the island value in the congenital spatial weight.
The new W object still causes the same problem I used to get before, I am not sure where the problem comes from.
Here is the piece of code

''

`shape=ps.open('file.dbf')`
`shape=shape.by_col_array('Solar')`
`shape=shape.astype(float)`
`hape[np.isnan(shape)]=0`
`scaler=MinMaxScaler()`
`shape1=scaler.fit_transform(shape)`
`w=ps.weights.Queen.from_shapefile('file.shp')`
`knn5=ps.knnW_from_shapefile('file.shp')`
`aa=w.islands`
`mtrx,idx=w.full()`
`for indx,val in enumerate(aa):`

        w.neighbors[aa[indx]]=[knn5.neighbors[aa[indx]][0]]
        w.neighbors[knn5.neighbors[aa[indx]][0]]=w.neighbors[knn5.neighbors[aa[indx]][0]]+[aa[indx]]

`w1=ps.weights.weights.W(w.neighbors,id_order=w.id_order,ids=w.id_order)`
`thr=0.1*sum(shape1[:,0])`
`np.random.seed(1234)`
`r=ps.region.maxp.Maxp(w1,shape1,floor=thr,floor_variable=shape1[:,0],initial=4000)`

Then I get the command
''No initial solution found''

  
''
Any help please?
